Code/Framework/AzToolsFramework/AzToolsFramework/UI/DPEDebugViewer/DPEDebugModel.cpp

```diff
@@ -113,36 +119,118 @@ namespace AzToolsFramework
+             ++columnIterator)
+        {
+            if ((*columnIterator)->GetDomValueIndex() == domIndex)
+            {
+                returnedNode = *columnIterator;
+                break;
+            }
+        }
+
+        return returnedNode;
+    }
+
     int DPEModelNode::RowOfChild(DPEModelNode* const childNode) const
     {
-        return m_children.indexOf(childNode);
+        return m_rowChildren.indexOf(childNode);
     }
 
-    void DPEModelNode::Populate(const AZ::Dom::Value& domVal)
+    int DPEModelNode::ColumnOfChild(DPEModelNode* const childNode) const
     {
-        auto& jsonBackend = GetModel()->GetBackend();
-        for (size_t arrayIndex = 0, numIndices = domVal.ArraySize(); arrayIndex < numIndices; ++arrayIndex)
+        return m_columnChildren.indexOf(childNode);
+    }
+
+    void DPEModelNode::SetValue(const AZ::Dom::Value& domVal, bool notifyView)
+    {
+        AZStd::string domName = domVal.GetNodeName().GetCStr();
+
+        // these will only be used if we're notifying the view
+        DPEDebugModel* theModel = nullptr;
+        QPersistentModelIndex myIndex;
+        if (notifyView)
         {
-            auto& currentValue = domVal[arrayIndex];
-            AZStd::string currentName = currentValue.GetNodeName().GetCStr();
+            theModel = GetModel();
+            myIndex = theModel->GetIndexFromNode(this);
+        }
 
-            // "Row" children start new rows, and are themselves populated, other node types are additional colu
-            if (currentName == AZ::DocumentPropertyEditor::Nodes::Row::Name)
+        // clear any children before repopulating
+        if (!m_rowChildren.isEmpty() || !m_columnChildren.isEmpty())
+        {
+            if (notifyView && !m_rowChildren.isEmpty())
             {
-                DPEModelNode* newRow = AddChild(NodeType::RowNode, arrayIndex, false, QString::number(m_children.count()));
-                newRow->Populate(currentValue);
+                theModel->beginRemoveRows(myIndex, 0, GetRowChildCount() - 1);
+                ClearChildren();
+                theModel->endRemoveRows();
             }
-            else if (currentName == AZ::DocumentPropertyEditor::Nodes::Label::Name)
+            else
+            {
+                ClearChildren();
+            }
+        }
+
+        // populate type and string, if applicable
+        if (domName == AZ::DocumentPropertyEditor::Nodes::Adapter::Name)
+        {
+            m_type = NodeType::RootNode;
+        }
+        else if (domName == AZ::DocumentPropertyEditor::Nodes::Row::Name)
+        {
+            m_type = NodeType::RowNode;
+        }
+        else if (domName == AZ::DocumentPropertyEditor::Nodes::Label::Name)
+        {
+            m_type = NodeType::LabelNode;
+            m_displayString =
+                QString::fromUtf8(AZStd::string(domVal[AZ::DocumentPropertyEditor::Nodes::Label::Value.GetName()].GetString()).c_str());
```

Comment:
Why convert `domVal[AZ::DocumentPropertyEditor::Nodes::Label::Value.GetName()].GetString()` to an AZStd::string to then convert it to a String?
Why not just go directly to QString?

Reply:
good call, fixed.